recipes/qt/6.x.x/conanfile.py

```diff
@@ -283,8 +283,12 @@ def validate(self):
 
             if not (self.options.gui and self.options.qtdeclarative and self.options.qtwebchannel):
                 raise ConanInvalidConfiguration("option qt:qtwebengine requires also qt:gui, qt:qtdeclarative and qt:qtwebchannel")
-            if not self.options.with_dbus and self.settings.os == "Linux":
-                raise ConanInvalidConfiguration("option qt:webengine requires also qt:with_dbus on Linux")
+            if self.settings.os == "Linux":
+                if not self.options.with_dbus:
+                    raise ConanInvalidConfiguration("option qt:webengine requires also qt:with_dbus on Linux")
+                if not self.options.with_glib:
+                    # it's not really needed, it's only for the pdf reader inside of chromium
+                    raise ConanInvalidConfiguration("option qt:webengine requires also qt:with_glib on Linux")
```

Comment:
if it's not actually required, please don't raise an exception. It will break some users which don't care about pdf reader.
Instead, you could either print a warning message, or add an option to the recipe for the pdf reader.

Reply:
I understand.
At the moment, the qtpdf is automatically brought in when compiling the qtwebengine and I don't know how to explicitly disable it.  Since it's automatically brought it, it does make it a required component in that sense, but if there was a away to disable qtpdf, then it would become optional.

Reply:
My impression has been that qtpdf is a really portion of qtwebengine.

Reply:
I can see why you say that.
So just to confirm that I understand, the suggestion is that I expose the qtpdf module as a build option in conan add that extra check in the validatiion for glib?